emcc.py

```diff
@@ -2080,9 +2074,12 @@ def compile_source_file(i, input_file):
         if shared.Settings.WASM_BACKEND and shared.Settings.RELOCATABLE:
           cmd.append('-fPIC')
           cmd.append('-fvisibility=default')
-        if shared.Settings.WASM_OBJECT_FILES:
-          for a in shared.Building.llvm_backend_args():
-            cmd += ['-mllvm', a]
+        if shared.Settings.WASM_BACKEND:
+          if shared.Settings.LTO:
+            cmd.append('-flto=' + shared.Settings.LTO)
+          else:
+            for a in shared.Building.llvm_backend_args():
+              cmd += ['-mllvm', a]
```

Comment:
don't we need these flags for LTO as well? i.e. shouldn't we append an extra `-flto` flag, but not remove the other stuff in `llvm_backend_args()`? perhaps we can even add that extra flag in that function? I may be missing something...

Reply:
Oops, you are completely correct.

Reply:
Oh.. with the old path we only pass the `-mllvm` options to the compiler if we are building object files.  Those flags are only used back the codegen/backend I guess.. which doesn't run until link with with LTO.

I guess there is no harm in adding them here since I'm not sure that all the flags are codegen-only, but it would make this change more impactful.  The change as it stands emulates the current behaviour.

Reply:
sgtm, maybe a TODO to investigate though?